SpeechRecognition/app/src/main/java/org/pytorch/demo/speechrecognition/MainActivity.java

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2020 Facebook, Inc. and its affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+// Copyright (c) 2020 Facebook, Inc. and its affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+
+package org.pytorch.demo.speechrecognition;
+
+import android.content.Context;
+import android.media.AudioFormat;
+import android.media.AudioRecord;
+import android.media.MediaRecorder;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import org.pytorch.IValue;
+import org.pytorch.Module;
+import org.pytorch.Tensor;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.FloatBuffer;
+import java.util.Map;
+
+
+class ListeningActivity extends AppCompatActivity implements Runnable {
+    public final static int AUDIO_LEN_LIMIT = 6;
+    private Button mButton;
+
+    public ListeningActivity(Button button) {
+        mButton = button;
+    }
+
+    public void run() {
+        for (int i = 1; i < AUDIO_LEN_LIMIT; i++) {
+            try {
+                Thread.sleep(1000);
```

Comment:
Using thread sleep is not androidish.
I would switch it to using Handler and HandlerThread as we use in https://github.com/pytorch/pytorch/blob/master/android/test_app/app/src/main/java/org/pytorch/testapp/CameraActivity.java:
```
HandlerThread mBackgroundThread;
Handler mBackgroundHandler;
Runnable mRunnable = new Runnable() {
      @Override
      public void run() {

        // Do something

        mBackgroundHandler.postDelayed(mRunnable, 1000);
      }
    };

mBackgroundThread = new HandlerThread("ModuleActivity");
mBackgroundThread.start();
mBackgroundHandler = new Handler(mBackgroundThread.getLooper());

mBackgroundHandler.postDelayed(mRunnable, 1000);
```